src/offset_parser.rs

```diff
@@ -0,0 +1,20 @@
+// #![allow(warnings)]
+use pulldown_cmark::{Event};
+
+pub struct OffsetParser<'a>(pub ::pulldown_cmark::Parser<'a>);
+
+impl<'a> Iterator for OffsetParser<'a> {
+    type Item = (Event<'a>, usize);
+
+    fn next(&mut self) -> Option<(Event<'a>, usize)> {
+        let &mut OffsetParser(ref mut inner_iter) = self;
+        let offset = inner_iter.get_offset();
+
+        match inner_iter.next() {
+            Some(event) => {
+                Some((event, offset))
```

Comment:
Any reason not to return `Some(WithOffset::new(event, offset))` here (and let that change trickle down)? If you make its fields public you can destructure it like a tuple. (Or make it a tuple struct?)